### PR TITLE
MultiRoot Certificate support: e2e tests

### PR DESCRIPTION
**Pull request by shankgan**
Adding preliminary security e2e tests to communicate between workloads signed by different root certificates. Refer to [RFC](https://docs.google.com/document/d/1EFMQZNlyImLP0k0HIyoBINntchqeUq7QQgc9O_jDwZ8/edit#heading=h.io2z16q0yjxj)

Related to https://github.com/istio/istio/pull/31009

Related to [Issue](https://github.com/istio/istio/issues/31111)

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by shankgan** [comment]
@jacob-delgado - Please review if possible

File: tests/integration/security/ca_custom_root/main_test.go [hunk 1]
```diff
@@ -271,9 +321,50 @@ func setupConfig(_ resource.Context, cfg *istio.Config) {
...
+	//	return
+	//}
+
+	cfgYaml := `
 values:
   meshConfig:
     trustDomainAliases: [some-other, trust-domain-foo]
+    caCertificates:
+    - pem: |
+        -----BEGIN CERTIFICATE-----
```
Comment (howardjohn) [inline tests/integration/security/ca_custom_root/main_test.go:339]: can we get this from whatever file its from?

**Review by howardjohn — COMMENTED** [review]
neat

File: tests/integration/security/ca_custom_root/main_test.go (hunk 1 above)
Comment (shankgan) [reply tests/integration/security/ca_custom_root/main_test.go:339]: working on it..not able to find a straightforward way to add multiline strings to yaml in golang

Comment (howardjohn) [reply tests/integration/security/ca_custom_root/main_test.go:339]: oh yeah the indent is annoying - i remember that. You could use `tmpl.MustEvaluate` then `{{.cert | indent 6}}` or similar though

**Review by jacob-delgado — APPROVED** [review]
This is a good use case that I don't think a lot of people are aware of doing.

**Review by jacob-delgado — CHANGES_REQUESTED** [review]
Missed the failing integration test. It may have to do with some of the setup, but I think this can be done.

**Comment by howardjohn** [comment]
this is expected to fail until the other pr is merged, right?

File: tests/integration/security/ca_custom_root/main_test.go (hunk 1 above)
Comment (shankgan) [reply tests/integration/security/ca_custom_root/main_test.go:339]: cool, will try this

**Comment by shankgan** [comment]
> this is expected to fail until the other pr is merged, right?

yeah, this is still waiting for https://github.com/istio/istio/pull/31009 to be approved and merged.

**Review by jacob-delgado — APPROVED** [review]
You should address Yangmins comments. But looks good.

File: tests/integration/security/ca_custom_root/multi_root_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+					serverNakedFooAlt := apps.ServerNakedFooAlt.GetOrFail(ctx, echo.InCluster(cluster))
+					cases := []struct {
+						src    echo.Instance
+						dest   echo.Instance
+						expect bool
+					}{
+						{
+							src:    client,
+							dest:   serverNakedFooAlt,
+							expect: true,
```
[the remark this answers is not in the document]
Comment (shankgan) [reply tests/integration/security/ca_custom_root/multi_root_test.go:76]: will add this in the subsequent PR

**Review by ericvn — COMMENTED** [review]
LGTM. You should address Yangmin's and John's comments.

File: samples/certs/README.md [hunk 3]
```diff
@@ -7,8 +7,11 @@ Instructions are available [here](https://istio.io/docs/tasks/security/cert-mana
 The included sample files are:
 
 - `root-cert.pem`: root CA certificate.
+- `root-cert-alt.pem`: alterative CA certificate.
```
[the remark this answers is not in the document]
Comment (shankgan) [reply samples/certs/README.md:10]: done

**Review by myidpt — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Pull request merged**